### PR TITLE
Added RandomHeight as native KerasCV preprocessing layer

### LINKED ISSUE #468: Migrate RandomHeight KPL to Keras_CV

### PR DESCRIPTION
**Pull request by suvadityamuk**
# What does this PR do?

Fixes #468 and adds RandomHeight layer to be used for preprocessing. Necessary tests for the code have also been added with all of them passing. Ported/Migrated from `keras/layers/preprocessing/image_preprocessing.py` and its respective tests.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
[Link to RandomHeight Discussion](https://github.com/keras-team/keras-cv/issues/468)
- [X] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@LukeWood @tanzhenyu

### CODE REVIEW

**Comment by suvadityamuk** [comment]
@bhack requesting a review

**Comment by suvadityamuk** [comment]
@LukeWood @tanzhenyu Any updates on the review?

**Comment by suvadityamuk** [comment]
Hi, folks! Any updates on the review? @LukeWood @tanzhenyu @bhack

**Comment by suvadityamuk** [comment]
Hey there, anything I need to update in the PR? Let me know! @LukeWood @tanzhenyu @bhack

File: examples/layers/preprocessing/classification/random_height_demo.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"""
+random_height_demo.py shows how to use the RandomHeight preprocessing layer for
+object detection.
+"""
+import demo_utils
```
Comment (tanzhenyu) [inline examples/layers/preprocessing/classification/random_height_demo.py:5]: Sharing the same set of comments with the other RandomWidth PR

**Review by tanzhenyu — CHANGES_REQUESTED** [review]

File: examples/layers/preprocessing/classification/random_height_demo.py (hunk 1 above)
Comment (suvadityamuk) [reply examples/layers/preprocessing/classification/random_height_demo.py]: Right, okay. Have removed it!

**Comment by suvadityamuk** [comment]
@LukeWood @bhack @tanzhenyu Hi, any updates on the review?

**Comment by divyashreepathihalli** [comment]
@suvadityamuk  - can you please add a demo file for RandomHeight?
here are some example
- classification demo - https://github.com/keras-team/keras-cv/blob/master/examples/layers/preprocessing/classification/random_crop_demo.py
- Bounding box demo - https://github.com/keras-team/keras-cv/blob/master/examples/layers/preprocessing/bounding_box/random_rotation_demo.py

Also please add the demo output here.

**Review by divyashreepathihalli — CHANGES_REQUESTED** [review]
@suvadityamuk - can you please add a demo file for RandomHeight?
here are some examples

classification demo - https://github.com/keras-team/keras-cv/blob/master/examples/layers/preprocessing/classification/random_crop_demo.py
Bounding box demo - https://github.com/keras-team/keras-cv/blob/master/examples/layers/preprocessing/bounding_box/random_rotation_demo.py
Also please add the demo output here.

File: keras_cv/layers/__init__.py [hunk 2]
```diff
@@ -13,7 +13,10 @@
 # limitations under the License.
 
 from tensorflow.keras.layers import CenterCrop
-from tensorflow.keras.layers import RandomHeight
+from tensorflow.keras.layers import RandomBrightness
```
Comment (LukeWood) [inline keras_cv/layers/__init__.py:16]: lets revert these spurious changes

**Comment by LukeWood** [comment]
Closing due to inactivity.  Thanks for the PR~.

**Pull request closed without merging**